File: ws/gavilanez/u2/ws21Inherence/Zoo/src/main/java/ec/edu/espe/zoo/model/Feline.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.zoo.model;

import java.util.Date;

/**
 *
 * @author Kenny Gavilanez, Dev Dynasty, DCCO-ESPE
 */
public abstract class Feline extends Animal{
    
    public Feline(int id, String description, Date dateOnborn, char gender, int numberOfBones, Cage cage) {
        super(id, description, dateOnborn, gender, numberOfBones, cage);
    }
    
    
    
}

File: hw/Vaca/u2/HW18ZooInPythonJavaScript/Zoo/ec.edu.espe.zoo.model/Feline.js
const Mammal = require('./Mammal');

class Feline extends Mammal {
    constructor(id, description, dateOnborn, gender, numberOfBones, cage, furColor) {
        super(id, description, dateOnborn, gender, numberOfBones, cage);
        this.furColor = furColor;
    }

    getFurColor() {
        return this.furColor;
    }

    setFurColor(furColor) {
        this.furColor = furColor;
    }

    toString() {
        return `Feline{furColor=${this.furColor}${super.toString()}}`;
    }
}

module.exports = Feline;
